stdlib/importlib/metadata/_meta.pyi

```diff
@@ -12,11 +13,11 @@ class PackageMetadata(Protocol):
     @property
     def json(self) -> dict[str, str | list[str]]: ...
 
+# Protocol deviates from runtime implementation.  The latter is not assignable
+# to common path-likes like `pathlib.Path` and `zipfile.Path`.
 class SimplePath(Protocol):
-    def joinpath(self) -> SimplePath: ...
-    def parent(self) -> SimplePath: ...
-    def read_text(self) -> str: ...
-    # There was a bug in `SimplePath` definition in cpython, see #8451
-    #  Strictly speaking `__div__` was defined in 3.10, not __truediv__,
-    # but it should have always been `__truediv__`.
-    def __truediv__(self) -> SimplePath: ...
+    def joinpath(self: Self, *other: str | Self) -> Self: ...
```

Comment:
I would not use `Self` here: that makes the protocol stricter than necessary. Instead, I would only require that it returns an instance of `SimplePath`.